core/data/tests/generate_types/output.ts

```diff
@@ -8,6 +8,7 @@ export interface Types {
 	float: number;
 	double: number;
 	array: string[];
+	fixed_length_array: [string, string, string, string];
```

Comment:
This is idiomatic, though I do worry that large values of `n` could blow up the file size significantly. I wonder if we could conditionally define an n-length tuple like so:

```ts
type FixedArray<T, N extends number> = N extends N ? number extends N ? T[] : _FixedArrayOf<T, N, []> : never;
type _FixedArrayOf<T, N extends number, R extends unknown[]> = R['length'] extends N ? R : _FixedArrayOf<T, N, [T, ...R]>;
```
and then use `Tuple<string, 4>`.

(taken from [this Stack Overflow answer](https://stackoverflow.com/a/52490977))

I can work on this in a follow-up.